### PR TITLE
Enable anisotropic filtering only beyond 20 degrees pitch

### PR DESCRIPTION
**Pull request by stepankuzmin**
Enable anisotropic filtering on tiles beyond 20 degrees pitch to prevent it from compromising image crispness on flat or low-tilted maps. This is a backport from GL Native. Closes https://github.com/mapbox/mapbox-gl-js/issues/7331

https://github.com/mapbox/mapbox-gl-js/blob/a3ef3657c2d0bff1c9d8442c5fd3fa7ded63b9f0/src/render/draw_raster.js#L81-L85

Also:
* Update Docker Image to the latest Node v14
* Lower the baseline for some render tests mostly because Firefox results are slightly different from Chrome in some tests
* Bumps `puppeteer-core` and `selenium-webdriver` to the latest versions

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Enable anisotropic filtering on tiles beyond 20 degrees pitch to prevent it from compromising image crispness on flat or low-tilted maps.</changelog>`

### CODE REVIEW

File: src/source/tile.js [hunk 1]
```diff
@@ -606,7 +606,12 @@ class Tile {
             this.texture = new Texture(context, img, gl.RGBA, {useMipmap: true});
             this.texture.bind(gl.LINEAR, gl.CLAMP_TO_EDGE);
 
-            if (context.extTextureFilterAnisotropic) {
+            // if greater it will mostly use mag filter
+            const mismatchingSizes = this.tileSize < img.width;
+
+            // Enable trilinear filtering on tiles only beyond 20 degrees pitch,
+            // to prevent it from compromising image crispness on flat or low tilted maps.
+            if (context.extTextureFilterAnisotropic && (painter.transform.pitch > 20 || mismatchingSizes)) {
```
[the remark this answers is not in the document]
Comment (stepankuzmin) [reply src/source/tile.js:614]: We call `setTexture` only after the tile is loaded in `RasterTileSource`, so it's worth checking. Thanks!

[the remark this answers is not in the document]
Comment (stepankuzmin) [reply src/source/tile.js]: I've moved it to the `drawRaster` call https://github.com/mapbox/mapbox-gl-js/pull/12577/commits/e1149bf0fa641200e99f2897f1eb2c63c1afcc66

**Review by mourner — APPROVED** [review]
Looks good to me! One final thing I'd check just in case is that the filter on draw doesn't incur any performance penalties, but otherwise good to go. Not sure what's going on with the CodeBuild failing but I guess that's a fluke.

**Comment by stepankuzmin** [comment]
I've run a benchmap, and it seems that the fix doesn't affect the rendering performance https://sites.mapbox.com/benchmap-js-results/runs/1358

**Pull request merged**